src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java

```diff
@@ -77,6 +79,21 @@ public void put(Collection<SinkRecord> records) {
         for (Map.Entry<String, List<SinkRecord>> entry : recordsByContainer.entrySet()) {
             String containerName = entry.getKey();
             CosmosContainer container = client.getDatabase(config.getDatabaseName()).getContainer(containerName);
+
+            // get the writer for this container
+            IWriter cosmosdbWriter = this.containerWriterMap.compute(containerName, (name, writer) -> {
+                if (writer == null) {

```

Comment:
Can writer be different for different containers?
May be not right now, but could be an interesting feature for future. 
Are we exposing enable bulk mode on sink connector on each container or for all containers?

Reply:
writer can be different for different containers -- this is the current implementation

and we are exposing the bulk mode per sink connector, so for all container. Created a working item for tracking: https://github.com/microsoft/kafka-connect-cosmosdb/issues/486